src/renderer/api/kube-object.ts

```diff
@@ -55,12 +56,233 @@ export class KubeObject implements ItemObject {
     return !item.metadata.name.startsWith("system:");
   }
 
-  static isJsonApiData(object: any): object is KubeJsonApiData {
-    return !object.items && object.metadata;
+  static isJsonApiData(object: unknown): object is KubeJsonApiData {
+    if (!object || typeof object !== "object") {
+      return false;
+    }
+
+    if (!hasOwnProperties(object, "kind", "apiVersion", "metadata")) {
+      return false;
+    }
+
+    if (typeof object.kind !== "string"
+      || typeof object.apiVersion !== "string"
+      || !object.metadata || typeof object.metadata !== "object"
+    ) {
+      return false;
+    }
+
+    const { metadata } = object;
+
+    if (!hasOwnProperties(metadata, "uid", "name", "resourceVersion", "selfLink")) {
+      return false;
+    }
+
+    if (typeof metadata.uid !== "string"
+      || typeof metadata.name !== "string"
+      || typeof metadata.resourceVersion !== "string"
+      || typeof metadata.selfLink !== "string"
+    ) {
+      return false;
+    }
+
+    if (hasOwnProperty(metadata, "namespace")
+      && typeof metadata.namespace !== "string"
+      && typeof metadata.namespace !== "undefined"
+    ) {
+      return false;
+    }
+
+    if (hasOwnProperty(metadata, "creationTimestamp")
+      && typeof metadata.creationTimestamp !== "string"
+      && typeof metadata.creationTimestamp !== "undefined"
+    ) {
+      return false;
+    }
+
+    if (hasOwnProperty(metadata, "continue")
+      && typeof metadata.continue !== "string"
+      && typeof metadata.continue !== "undefined"
+    ) {
+      return false;
+    }
+
+    if (hasOwnProperty(metadata, "finalizers")
+      && typeof metadata.finalizers !== "undefined"
+      && !(Array.isArray(metadata.finalizers)
+      && metadata.finalizers.every(finalizer => typeof finalizer === "string"))
+    ) {
+      return false;
+    }
+
+    labels: if (hasOwnProperty(metadata, "labels")) {
+      if (typeof metadata.labels === "undefined" || (typeof metadata.labels === "object" && metadata.labels !== null && !Array.isArray(metadata.labels))) {
+        if (metadata.labels) {
+          for (const label in metadata.labels) {
+            if (!hasOwnProperty(metadata.labels, label) || typeof metadata.labels[label] !== "string") {
+              return false;
+            }
+          }
```

Comment:
Would `Object.entries(metadata.labels).forEach(...)` eliminate the need to use `hasOwnProperty()`?

Reply:
This check technically verifies that there the values are not "up the prototype" chain. Furtherore, we cannot use `forEach` we would have to use `every` again because you cannot return out of several functions at once.

Reply:
I know, but it's 2021, we still have to rely on `hasOwnProperty`...